Kitodo/src/main/java/org/kitodo/production/services/file/FileService.java

```diff
@@ -70,6 +69,8 @@
 
     private static final Logger logger = LogManager.getLogger(FileService.class);
 
+    public final MetadataImageComparator metadataImageComparator = new MetadataImageComparator(this);
```

Comment:
Public class variables? Use a getter method for accessing this variable and implement the used class as a singleton then you did not need this.

Reply:
Singleton isn’t possible here, because of back-reference to non-static file service